// @atlaskit/motion/dist/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ExitingPersistence", {
  enumerable: true,
  get: function get() {
    return _exitingPersistence.default;
  }
});
Object.defineProperty(exports, "FadeIn", {
  enumerable: true,
  get: function get() {
    return _fadeIn.default;
  }
});
Object.defineProperty(exports, "ResizingHeight", {
  enumerable: true,
  get: function get() {
    return _height.ResizingHeight;
  }
});
Object.defineProperty(exports, "ShrinkOut", {
  enumerable: true,
  get: function get() {
    return _shrinkOut.default;
  }
});
Object.defineProperty(exports, "SlideIn", {
  enumerable: true,
  get: function get() {
    return _slideIn.default;
  }
});
Object.defineProperty(exports, "StaggeredEntrance", {
  enumerable: true,
  get: function get() {
    return _staggeredEntrance.default;
  }
});
Object.defineProperty(exports, "ZoomIn", {
  enumerable: true,
  get: function get() {
    return _zoomIn.default;
  }
});
Object.defineProperty(exports, "durationStep", {
  enumerable: true,
  get: function get() {
    return _durations.durationStep;
  }
});
Object.defineProperty(exports, "easeIn", {
  enumerable: true,
  get: function get() {
    return _curves.easeIn;
  }
});
Object.defineProperty(exports, "easeInOut", {
  enumerable: true,
  get: function get() {
    return _curves.easeInOut;
  }
});
Object.defineProperty(exports, "easeOut", {
  enumerable: true,
  get: function get() {
    return _curves.easeOut;
  }
});
Object.defineProperty(exports, "fadeInAnimation", {
  enumerable: true,
  get: function get() {
    return _fadeIn.fadeInAnimation;
  }
});
Object.defineProperty(exports, "isReducedMotion", {
  enumerable: true,
  get: function get() {
    return _accessibility.isReducedMotion;
  }
});
Object.defineProperty(exports, "largeDurationMs", {
  enumerable: true,
  get: function get() {
    return _durations.largeDurationMs;
  }
});
Object.defineProperty(exports, "mediumDurationMs", {
  enumerable: true,
  get: function get() {
    return _durations.mediumDurationMs;
  }
});
Object.defineProperty(exports, "prefersReducedMotion", {
  enumerable: true,
  get: function get() {
    return _accessibility.prefersReducedMotion;
  }
});
Object.defineProperty(exports, "shrinkOutAnimation", {
  enumerable: true,
  get: function get() {
    return _zoomIn.shrinkOutAnimation;
  }
});
Object.defineProperty(exports, "slideInAnimation", {
  enumerable: true,
  get: function get() {
    return _slideIn.slideInAnimation;
  }
});
Object.defineProperty(exports, "smallDurationMs", {
  enumerable: true,
  get: function get() {
    return _durations.smallDurationMs;
  }
});
Object.defineProperty(exports, "useExitingPersistence", {
  enumerable: true,
  get: function get() {
    return _exitingPersistence.useExitingPersistence;
  }
});
Object.defineProperty(exports, "useIsReducedMotion", {
  enumerable: true,
  get: function get() {
    return _accessibility.useIsReducedMotion;
  }
});
Object.defineProperty(exports, "useResizingHeight", {
  enumerable: true,
  get: function get() {
    return _height.useResizingHeight;
  }
});
Object.defineProperty(exports, "zoomInAnimation", {
  enumerable: true,
  get: function get() {
    return _zoomIn.zoomInAnimation;
  }
});

var _durations = require("./utils/durations");

var _curves = require("./utils/curves");

var _accessibility = require("./utils/accessibility");

var _fadeIn = _interopRequireWildcard(require("./entering/fade-in"));

var _staggeredEntrance = _interopRequireDefault(require("./entering/staggered-entrance"));

var _height = require("./resizing/height");

var _exitingPersistence = _interopRequireWildcard(require("./entering/exiting-persistence"));

var _zoomIn = _interopRequireWildcard(require("./entering/zoom-in"));

var _slideIn = _interopRequireWildcard(require("./entering/slide-in"));

var _shrinkOut = _interopRequireDefault(require("./entering/shrink-out"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
