// @atlaskit/motion/dist/es2019/entering/shrink-out.js
import { easeIn } from '../utils/curves';
import { smallDurationMs } from '../utils/durations';
import { useRequestAnimationFrame, useSetTimeout } from '../utils/timer-hooks';
import { useElementRef } from '../utils/use-element-ref';
import { useLayoutEffect } from '../utils/use-layout-effect';
import { useExitingPersistence } from './exiting-persistence';

const ShrinkOut = ({
  children,
  duration = smallDurationMs,
  onFinish
}) => {
  const [element, setElementRef] = useElementRef();
  const exiting = useExitingPersistence();
  const requestAnimationFrame = useRequestAnimationFrame();
  const setTimeout = useSetTimeout();
  const direction = exiting.isExiting ? 'exiting' : 'entering';
  useLayoutEffect(() => {
    if (exiting.isExiting && element) {
      const newStyles = {
        // We fix both width and height because when changing box sizing to border-box.
        width: `${element.offsetWidth}px`,
        height: `${element.offsetHeight}px`,
        boxSizing: 'border-box',
        willChange: 'width,margin'
      };
      Object.assign(element.style, newStyles);
      requestAnimationFrame(() => {
        requestAnimationFrame(() => {
          const newStyles = {
            width: '0px',
            margin: '0px',
            // We animate margin down to zero so it doesn't take any space.
            transitionTimingFunction: easeIn,
            transitionDuration: `${duration}ms`,
            transitionProperty: 'width,margin'
          };
          Object.assign(element.style, newStyles);
          setTimeout(() => {
            exiting.onFinish && exiting.onFinish();
            onFinish && onFinish('exiting');
          }, duration);
        });
      });
    }
  });
  return children({
    ref: setElementRef
  }, direction);
};

export default ShrinkOut;
