// @atlaskit/codemod-utils/dist/cjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "addCommentBefore", {
  enumerable: true,
  get: function get() {
    return _support.addCommentBefore;
  }
});
Object.defineProperty(exports, "addCommentToStartOfFile", {
  enumerable: true,
  get: function get() {
    return _support.addCommentToStartOfFile;
  }
});
Object.defineProperty(exports, "addDynamicImport", {
  enumerable: true,
  get: function get() {
    return _support.addDynamicImport;
  }
});
Object.defineProperty(exports, "addToImport", {
  enumerable: true,
  get: function get() {
    return _support.addToImport;
  }
});
Object.defineProperty(exports, "callExpressionArgMatchesString", {
  enumerable: true,
  get: function get() {
    return _support.callExpressionArgMatchesString;
  }
});
Object.defineProperty(exports, "changeImportEntryPoint", {
  enumerable: true,
  get: function get() {
    return _utils.changeImportEntryPoint;
  }
});
Object.defineProperty(exports, "createConvertFuncFor", {
  enumerable: true,
  get: function get() {
    return _utils.createConvertFuncFor;
  }
});
Object.defineProperty(exports, "createRemoveFuncAddCommentFor", {
  enumerable: true,
  get: function get() {
    return _utils.createRemoveFuncAddCommentFor;
  }
});
Object.defineProperty(exports, "createRemoveFuncFor", {
  enumerable: true,
  get: function get() {
    return _utils.createRemoveFuncFor;
  }
});
Object.defineProperty(exports, "createRenameFuncFor", {
  enumerable: true,
  get: function get() {
    return _utils.createRenameFuncFor;
  }
});
Object.defineProperty(exports, "createRenameImportFor", {
  enumerable: true,
  get: function get() {
    return _utils.createRenameImportFor;
  }
});
Object.defineProperty(exports, "createRenameJSXFunc", {
  enumerable: true,
  get: function get() {
    return _utils.createRenameJSXFunc;
  }
});
Object.defineProperty(exports, "createTransformer", {
  enumerable: true,
  get: function get() {
    return _utils.createTransformer;
  }
});
Object.defineProperty(exports, "doesIdentifierExist", {
  enumerable: true,
  get: function get() {
    return _support.doesIdentifierExist;
  }
});
Object.defineProperty(exports, "elevateComponentToNewEntryPoint", {
  enumerable: true,
  get: function get() {
    return _utils.elevateComponentToNewEntryPoint;
  }
});
Object.defineProperty(exports, "flattenCertainChildPropsAsProp", {
  enumerable: true,
  get: function get() {
    return _utils.flattenCertainChildPropsAsProp;
  }
});
Object.defineProperty(exports, "getDefaultSpecifier", {
  enumerable: true,
  get: function get() {
    return _support.getDefaultSpecifier;
  }
});
Object.defineProperty(exports, "getDynamicImportName", {
  enumerable: true,
  get: function get() {
    return _support.getDynamicImportName;
  }
});
Object.defineProperty(exports, "getJSXAttributesByName", {
  enumerable: true,
  get: function get() {
    return _support.getJSXAttributesByName;
  }
});
Object.defineProperty(exports, "getNamedSpecifier", {
  enumerable: true,
  get: function get() {
    return _support.getNamedSpecifier;
  }
});
Object.defineProperty(exports, "getSafeImportName", {
  enumerable: true,
  get: function get() {
    return _support.getSafeImportName;
  }
});
Object.defineProperty(exports, "hasImportDeclaration", {
  enumerable: true,
  get: function get() {
    return _support.hasImportDeclaration;
  }
});
Object.defineProperty(exports, "hasImportDeclarationFromAnyPackageEntrypoint", {
  enumerable: true,
  get: function get() {
    return _support.hasImportDeclarationFromAnyPackageEntrypoint;
  }
});
Object.defineProperty(exports, "hasJSXAttributesByName", {
  enumerable: true,
  get: function get() {
    return _support.hasJSXAttributesByName;
  }
});
Object.defineProperty(exports, "removeImport", {
  enumerable: true,
  get: function get() {
    return _support.removeImport;
  }
});
Object.defineProperty(exports, "renameNamedImportWithAliasName", {
  enumerable: true,
  get: function get() {
    return _utils.renameNamedImportWithAliasName;
  }
});
Object.defineProperty(exports, "replaceImportStatementFor", {
  enumerable: true,
  get: function get() {
    return _utils.replaceImportStatementFor;
  }
});
Object.defineProperty(exports, "testMethodVariantEach", {
  enumerable: true,
  get: function get() {
    return _support.testMethodVariantEach;
  }
});
Object.defineProperty(exports, "tryCreateImport", {
  enumerable: true,
  get: function get() {
    return _support.tryCreateImport;
  }
});

var _utils = require("./utils");

var _support = require("./utils/support");
