// @atlaskit/motion/dist/cjs/entering/fade-in.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.fadeOutAnimation = exports.fadeInAnimation = exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _curves = require("../utils/curves");

var _durations = require("../utils/durations");

var _keyframesMotion = _interopRequireDefault(require("./keyframes-motion"));

var _excluded = ["children", "duration", "entranceDirection"];

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2.default)(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

var entranceMotions = {
  bottom: 'translate3d(0, calc(5% + 4px), 0)',
  left: 'translate3d(calc(-5% - 4px), 0, 0)',
  right: 'translate3d(calc(5% + 4px), 0, 0)',
  top: 'translate3d(0, calc(-5% - 4px), 0)'
};
var exitMotions = {
  bottom: 'translate3d(0, calc(-5% - 4px), 0)',
  left: 'translate3d(calc(5% + 4px), 0, 0)',
  right: 'translate3d(calc(-5% - 4px), 0, 0)',
  top: 'translate3d(0, calc(5% + 4px), 0)'
};

var fadeInAnimation = function fadeInAnimation(movement) {
  return {
    from: _objectSpread({
      opacity: 0
    }, movement !== undefined && {
      transform: entranceMotions[movement]
    }),
    '50%': {
      opacity: 1
    },
    to: {
      transform: movement !== undefined ? 'none' : undefined
    }
  };
};

exports.fadeInAnimation = fadeInAnimation;

var fadeOutAnimation = function fadeOutAnimation(movement) {
  return {
    from: {
      opacity: 1,
      transform: movement !== undefined ? 'translate3d(0, 0, 0)' : undefined
    },
    to: _objectSpread({
      opacity: 0
    }, movement !== undefined && {
      transform: exitMotions[movement]
    })
  };
};
/**
 * Props for controlling the behaviour of the FadeIn animation
 */


exports.fadeOutAnimation = fadeOutAnimation;

var FadeIn = function FadeIn(_ref) {
  var children = _ref.children,
      _ref$duration = _ref.duration,
      duration = _ref$duration === void 0 ? _durations.largeDurationMs : _ref$duration,
      entranceSlideDirection = _ref.entranceDirection,
      props = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  return /*#__PURE__*/_react.default.createElement(_keyframesMotion.default, (0, _extends2.default)({
    duration: duration,
    enteringAnimation: fadeInAnimation(entranceSlideDirection),
    exitingAnimation: fadeOutAnimation(entranceSlideDirection),
    animationTimingFunction: function animationTimingFunction() {
      return _curves.easeInOut;
    }
  }, props), children);
};

var _default = FadeIn;
exports.default = _default;
