// @atlaskit/motion/dist/cjs/entering/slide-in.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.slideInAnimation = exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireDefault(require("react"));

var _curves = require("../utils/curves");

var _durations = require("../utils/durations");

var _keyframesMotion = _interopRequireDefault(require("./keyframes-motion"));

var _excluded = ["children", "enterFrom", "exitTo", "fade", "duration", "animationTimingFunction"];

var slideInAnimation = function slideInAnimation(from, state, fade) {
  var _ref;

  var initial = state === 'entering' ? '0%' : '100%';
  var end = state === 'entering' ? '100%' : '0%';
  var fromMap = {
    top: 'translate3d(0, -100%, 0)',
    right: 'translate3d(100%, 0, 0)',
    bottom: 'translate3d(0, 100%, 0)',
    left: 'translate3d(-100%, 0, 0)'
  };
  var animateOpacity = fade === 'inout' || fade === 'in' && state === 'entering' || fade === 'out' && state === 'exiting';
  return _ref = {}, (0, _defineProperty2.default)(_ref, initial, {
    transform: fromMap[from],
    opacity: animateOpacity ? 0 : 1
  }), (0, _defineProperty2.default)(_ref, end, {
    transform: 'none',
    opacity: 1
  }), _ref;
};

exports.slideInAnimation = slideInAnimation;

var SlideIn = function SlideIn(_ref2) {
  var children = _ref2.children,
      enterFrom = _ref2.enterFrom,
      exitTo = _ref2.exitTo,
      _ref2$fade = _ref2.fade,
      fade = _ref2$fade === void 0 ? 'none' : _ref2$fade,
      _ref2$duration = _ref2.duration,
      duration = _ref2$duration === void 0 ? _durations.mediumDurationMs : _ref2$duration,
      _ref2$animationTiming = _ref2.animationTimingFunction,
      animationTimingFunction = _ref2$animationTiming === void 0 ? function (state) {
    return state === 'entering' ? _curves.easeOut : _curves.easeIn;
  } : _ref2$animationTiming,
      props = (0, _objectWithoutProperties2.default)(_ref2, _excluded);
  return /*#__PURE__*/_react.default.createElement(_keyframesMotion.default, (0, _extends2.default)({
    duration: duration,
    enteringAnimation: slideInAnimation(enterFrom, 'entering', fade),
    exitingAnimation: slideInAnimation(exitTo || enterFrom, 'exiting', fade),
    animationTimingFunction: animationTimingFunction
  }, props), children);
};

var _default = SlideIn;
exports.default = _default;
