// @atlaskit/motion/dist/cjs/entering/zoom-in.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.zoomInAnimation = exports.shrinkOutAnimation = exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _keyframesMotion = _interopRequireDefault(require("./keyframes-motion"));

var _excluded = ["children", "duration"];

var zoomInAnimation = function zoomInAnimation() {
  return {
    '0%': {
      opacity: 0,
      transform: 'scale(0.5)'
    },
    '50%': {
      opacity: 1
    },
    '75%': {
      transform: 'scale(1.25)'
    },
    '100%': {
      transform: 'scale(1)'
    }
  };
};

exports.zoomInAnimation = zoomInAnimation;

var shrinkOutAnimation = function shrinkOutAnimation() {
  return {
    to: {
      opacity: 0,
      transform: 'scale(0.75)'
    }
  };
};

exports.shrinkOutAnimation = shrinkOutAnimation;

var ZoomIn = function ZoomIn(_ref) {
  var children = _ref.children,
      _ref$duration = _ref.duration,
      duration = _ref$duration === void 0 ? 125 : _ref$duration,
      props = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  return /*#__PURE__*/_react.default.createElement(_keyframesMotion.default, (0, _extends2.default)({
    duration: duration,
    enteringAnimation: zoomInAnimation(),
    exitingAnimation: shrinkOutAnimation(),
    animationTimingFunction: function animationTimingFunction() {
      return 'ease-in-out';
    }
  }, props), children);
};

var _default = ZoomIn;
exports.default = _default;
