// @atlaskit/analytics-next/dist/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _typeof = require("@babel/runtime/helpers/typeof");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "AnalyticsContext", {
  enumerable: true,
  get: function get() {
    return _index2.default;
  }
});
Object.defineProperty(exports, "AnalyticsErrorBoundary", {
  enumerable: true,
  get: function get() {
    return _AnalyticsErrorBoundary.default;
  }
});
Object.defineProperty(exports, "AnalyticsEvent", {
  enumerable: true,
  get: function get() {
    return _AnalyticsEvent.default;
  }
});
Object.defineProperty(exports, "AnalyticsListener", {
  enumerable: true,
  get: function get() {
    return _index.default;
  }
});
Object.defineProperty(exports, "AnalyticsReactContext", {
  enumerable: true,
  get: function get() {
    return _analyticsNextStableReactContext.default;
  }
});
Object.defineProperty(exports, "UIAnalyticsEvent", {
  enumerable: true,
  get: function get() {
    return _UIAnalyticsEvent.default;
  }
});
Object.defineProperty(exports, "cleanProps", {
  enumerable: true,
  get: function get() {
    return _cleanProps.default;
  }
});
Object.defineProperty(exports, "createAndFireEvent", {
  enumerable: true,
  get: function get() {
    return _createAndFireEvent.default;
  }
});
Object.defineProperty(exports, "isAnalyticsEvent", {
  enumerable: true,
  get: function get() {
    return _AnalyticsEvent.isAnalyticsEvent;
  }
});
Object.defineProperty(exports, "isUIAnalyticsEvent", {
  enumerable: true,
  get: function get() {
    return _UIAnalyticsEvent.isUIAnalyticsEvent;
  }
});
Object.defineProperty(exports, "useAnalyticsEvents", {
  enumerable: true,
  get: function get() {
    return _useAnalyticsEvents.useAnalyticsEvents;
  }
});
Object.defineProperty(exports, "useCallbackWithAnalytics", {
  enumerable: true,
  get: function get() {
    return _useCallbackWithAnalytics.useCallbackWithAnalytics;
  }
});
Object.defineProperty(exports, "usePlatformLeafEventHandler", {
  enumerable: true,
  get: function get() {
    return _usePlatformLeafEventHandler.usePlatformLeafEventHandler;
  }
});
Object.defineProperty(exports, "usePlatformLeafSyntheticEventHandler", {
  enumerable: true,
  get: function get() {
    return _usePlatformLeafSyntheticEventHandler.usePlatformLeafSyntheticEventHandler;
  }
});
Object.defineProperty(exports, "withAnalyticsContext", {
  enumerable: true,
  get: function get() {
    return _withAnalyticsContext.default;
  }
});
Object.defineProperty(exports, "withAnalyticsEvents", {
  enumerable: true,
  get: function get() {
    return _withAnalyticsEvents.default;
  }
});

var _AnalyticsEvent = _interopRequireWildcard(require("./events/AnalyticsEvent"));

var _UIAnalyticsEvent = _interopRequireWildcard(require("./events/UIAnalyticsEvent"));

var _index = _interopRequireDefault(require("./components/AnalyticsListener/index"));

var _index2 = _interopRequireDefault(require("./components/AnalyticsContext/index"));

var _withAnalyticsContext = _interopRequireDefault(require("./hocs/withAnalyticsContext"));

var _AnalyticsErrorBoundary = _interopRequireDefault(require("./components/AnalyticsErrorBoundary"));

var _withAnalyticsEvents = _interopRequireDefault(require("./hocs/withAnalyticsEvents"));

var _analyticsNextStableReactContext = _interopRequireDefault(require("@atlaskit/analytics-next-stable-react-context"));

var _useAnalyticsEvents = require("./hooks/useAnalyticsEvents");

var _useCallbackWithAnalytics = require("./hooks/useCallbackWithAnalytics");

var _usePlatformLeafEventHandler = require("./hooks/usePlatformLeafEventHandler");

var _usePlatformLeafSyntheticEventHandler = require("./hooks/usePlatformLeafSyntheticEventHandler");

var _createAndFireEvent = _interopRequireDefault(require("./utils/createAndFireEvent"));

var _cleanProps = _interopRequireDefault(require("./utils/cleanProps"));

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
