// @atlaskit/tokens/dist/cjs/tokens/default/color/interaction.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var color = {
  color: {
    interaction: {
      hovered: {
        attributes: {
          group: 'paint',
          state: 'active',
          introduced: '0.6.0',
          description: 'Use as a background overlay for elements in a hovered state when their background color cannot change, such as avatars.'
        }
      },
      pressed: {
        attributes: {
          group: 'paint',
          state: 'active',
          introduced: '0.6.0',
          description: 'Use as a background overlay for elements in a pressed state when their background color cannot change, such as avatars.'
        }
      },
      inverse: {
        hovered: {
          attributes: {
            group: 'paint',
            state: 'active',
            introduced: '0.6.0',
            description: 'Use as a background overlay for elements in a hovered state on bold backgrounds, such as the buttons on spotlight cards.'
          }
        },
        pressed: {
          attributes: {
            group: 'paint',
            state: 'active',
            introduced: '0.6.0',
            description: 'Use as a background overlay for elements in a hovered state on bold backgrounds, such as the buttons on spotlight cards.'
          }
        }
      }
    }
  }
};
var _default = color;
exports.default = _default;
