// @atlaskit/tokens/dist/cjs/tokens/atlassian-light/elevation/shadow.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var shadow = {
  elevation: {
    shadow: {
      raised: {
        value: [{
          radius: 1,
          offset: {
            x: 0,
            y: 1
          },
          color: 'N1100',
          opacity: 0.25
        }, {
          radius: 1,
          offset: {
            x: 0,
            y: 0
          },
          color: 'N1100',
          opacity: 0.31
        }]
      },
      overflow: {
        value: [{
          radius: 8,
          offset: {
            x: 0,
            y: 8
          },
          color: 'N1100',
          opacity: 0.08
        }, {
          radius: 1,
          offset: {
            x: 0,
            y: 0
          },
          color: 'N1100',
          opacity: 0.12
        }]
      },
      overlay: {
        value: [{
          radius: 12,
          offset: {
            x: 0,
            y: 8
          },
          color: 'N1100',
          opacity: 0.15
        }, {
          radius: 1,
          offset: {
            x: 0,
            y: 0
          },
          color: 'N1100',
          opacity: 0.31
        }]
      }
    }
  }
};
var _default = shadow;
exports.default = _default;
