// @atlaskit/theme/dist/cjs/elevation.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.e500 = exports.e400 = exports.e300 = exports.e200 = exports.e100 = void 0;

var _colors = require("./colors");

var _themed = _interopRequireDefault(require("./utils/themed"));

/* eslint-disable @atlaskit/design-system/ensure-design-token-usage */
// Cards on a board
var e100 = (0, _themed.default)({
  light: "box-shadow: ".concat("var(--ds-shadow-raised, ".concat("0 1px 1px ".concat(_colors.N50A, ", 0 0 1px 1px ").concat(_colors.N40A), ")"), ";"),
  dark: "box-shadow: ".concat("var(--ds-shadow-raised, ".concat("0 1px 1px ".concat(_colors.DN50A, ", 0 0 1px 1px ").concat(_colors.DN40A), ")"), ";")
}); // Inline dialogs

exports.e100 = e100;
var e200 = (0, _themed.default)({
  light: "box-shadow: ".concat("var(--ds-shadow-overlay, ".concat("0 4px 8px -2px ".concat(_colors.N50A, ", 0 0 1px ").concat(_colors.N60A), ")"), ";"),
  dark: "box-shadow: ".concat("var(--ds-shadow-overlay, ".concat("0 4px 8px -2px ".concat(_colors.DN50A, ", 0 0 1px ").concat(_colors.DN60A), ")"), ";")
}); // Modals

exports.e200 = e200;
var e300 = (0, _themed.default)({
  light: "box-shadow: ".concat("var(--ds-shadow-overlay, ".concat("0 8px 16px -4px ".concat(_colors.N50A, ", 0 0 1px ").concat(_colors.N60A), ")"), ";"),
  dark: "box-shadow: ".concat("var(--ds-shadow-overlay, ".concat("0 8px 16px -4px ".concat(_colors.DN50A, ", 0 0 1px ").concat(_colors.DN60A), ")"), ";")
}); // Panels

exports.e300 = e300;
var e400 = (0, _themed.default)({
  light: "box-shadow: ".concat("var(--ds-shadow-overlay, ".concat("0 12px 24px -6px ".concat(_colors.N50A, ", 0 0 1px ").concat(_colors.N60A), ")"), ";"),
  dark: "box-shadow: ".concat("var(--ds-shadow-overlay, ".concat("0 12px 24px -6px ".concat(_colors.DN50A, ", 0 0 1px ").concat(_colors.DN60A), ")"), ";")
}); // Flag messages (notifications)

exports.e400 = e400;
var e500 = (0, _themed.default)({
  light: "box-shadow: ".concat("var(--ds-shadow-overlay, ".concat("0 20px 32px -8px ".concat(_colors.N50A, ", 0 0 1px ").concat(_colors.N60A), ")"), ";"),
  dark: "box-shadow: ".concat("var(--ds-shadow-overlay, ".concat("0 20px 32px -8px ".concat(_colors.DN50A, ", 0 0 1px ").concat(_colors.DN60A), ")"), ";")
});
exports.e500 = e500;
