// @atlaskit/motion/dist/esm/entering/zoom-in.js
import _extends from "@babel/runtime/helpers/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
var _excluded = ["children", "duration"];
import React from 'react';
import KeyframesMotion from './keyframes-motion';
export var zoomInAnimation = function zoomInAnimation() {
  return {
    '0%': {
      opacity: 0,
      transform: 'scale(0.5)'
    },
    '50%': {
      opacity: 1
    },
    '75%': {
      transform: 'scale(1.25)'
    },
    '100%': {
      transform: 'scale(1)'
    }
  };
};
export var shrinkOutAnimation = function shrinkOutAnimation() {
  return {
    to: {
      opacity: 0,
      transform: 'scale(0.75)'
    }
  };
};

var ZoomIn = function ZoomIn(_ref) {
  var children = _ref.children,
      _ref$duration = _ref.duration,
      duration = _ref$duration === void 0 ? 125 : _ref$duration,
      props = _objectWithoutProperties(_ref, _excluded);

  return /*#__PURE__*/React.createElement(KeyframesMotion, _extends({
    duration: duration,
    enteringAnimation: zoomInAnimation(),
    exitingAnimation: shrinkOutAnimation(),
    animationTimingFunction: function animationTimingFunction() {
      return 'ease-in-out';
    }
  }, props), children);
};

export default ZoomIn;
